cmd/provisioner-localpv/app/provisioner_hostpath.go

```diff
@@ -87,6 +93,43 @@ func (p *Provisioner) ProvisionHostPath(ctx context.Context, opts pvController.P
 		return nil, pvController.ProvisioningFinished, iErr
 	}
 
+	scType := opts.StorageClass.Parameters[ScType]
+
+	if scType == "enforceXfsQuota" {
+		bsoft := opts.StorageClass.Parameters[Bsoft]
+		bhard := opts.StorageClass.Parameters[Bhard]
+
+		podOpts := &HelperPodOptions{
+			name:                   name,
+			path:                   path,
+			nodeAffinityLabelKey:   nodeAffinityKey,
+			nodeAffinityLabelValue: nodeAffinityValue,
+			serviceAccountName:     saName,
+			selectedNodeTaints:     taints,
+			imagePullSecrets:       imagePullSecrets,
+			bsoft:                  bsoft,
+			bhard:                  bhard,
+		}
```

Comment:
All these can be performed outside of this if condition. If the fields are not filled by the user then they will remain empty. And for if condition you can check for:
```suggestion
		if scType == "enforceXfsQuota" && (len(Bsoft) != 0 || len(Bhard) != 0)
```
And then have proper error handling checks too.

Reply:
I don't think we should check like this `if scType == "enforceXfsQuota" && (len(Bsoft) != 0 || len(Bhard) != 0)
` because if user has given type `enforceXfsQuota` that means he wants to set quota , then we should try to create quota pod , but while pod creation the error `both limits can't be 0 ` will be displayed and it will fail and doing like this there will be no error displayed

Reply:
If we want to remove redeclaration of `podOpts` we can declare value of `bsoft` and `bhard` while sending it to helper pod , it will be empty anyways (if you don't give values in parameters )as variable is declared in `podOpts`.

Reply:
@almas33 - it is quite possible to misconfigure the parameters or provide incorrect parameters by users - while they are learning how to use the solution. The code should handle this via:
* Sane defaults
* Catch errors

Reply:
@kmova - does that mean there will be some default limits applied if there is error in limits entered in parameter and errors of incorrect parameters are already taken care in `convertToK()`

Reply:
@almas33, let us get back to this comment after the changes suggested in the design doc for storage class parameters are updated. The validation will change.